Makes `ap chat` prettier

raghubetina opened the pull request:
Makes `ap chat` and `pp chat` prettier.

## Changes

### Amazing Print Integration

- Added custom formatter (`lib/ai/amazing_print.rb`) that displays objects with proper notation instead of hash-like output.
- Objects now appear as `#<AI::Chat ...>` with instance variables shown using `@var:` syntax.
- Automatically loads when `amazing_print` is available.

### Pretty Print Support

- Implemented `pretty_print` methods for both `AI::Chat` and `AI::Response` classes.
- Works with Ruby's standard `pp` command out of the box.

### Features

- **Content truncation**: Long message content is automatically truncated to 80 characters with "..." for readability.
- **Security**: Sensitive instance variables (`@api_key`, `@client`) are hidden from output.
- **Clean formatting**: Proper indentation and object notation make debugging conversations easier.

## Example Output

**Before:**
```ruby
#<AI::Chat @messages=[{:role=>"system", :content=>"You are a helpful assistant"}, {:role=>"user", :content=>"What's the best pizza in Chicago?"}, {:role=>"assistant", :content=>"Hark! In fair Chicago..."}] @model="gpt-4.1-nano" @schema=nil @reasoning_effort=nil>
```

**After (with `ap`):**

<img width="1379" height="1051" alt="Screenshot 2025-07-30 at 2 16 33 PM" src="https://github.com/user-attachments/assets/5b83cb49-c836-4d4f-bc18-e9431f18ad9e" />


**After with `pp`:**

<img width="1796" height="558" alt="Screenshot 2025-07-30 at 2 16 19 PM" src="https://github.com/user-attachments/assets/3af755fa-da0d-4a42-8814-c5af7df9dbc1" />


## Usage

Simply use `pp` or `ap` as you normally would:

```ruby
require "ai-chat"
require "pp"  # or require "amazing_print"

chat = AI::Chat.new
chat.user("Hello!")
chat.generate!

pp chat  # or ap chat
```

## Other Changes

- Updated `.gitignore` to exclude temporary demo scripts.
- Minor dependency updates in `Gemfile.lock`.


<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Enhances `AI::Chat` and `AI::Response` output formatting with `amazing_print` and `pretty_print`, including content truncation and sensitive data hiding.
> 
>   - **Amazing Print Integration**:
>     - Adds `lib/ai/amazing_print.rb` for custom formatting of `AI::Chat` objects with `amazing_print`.
>     - Objects display as `#<AI::Chat ...>` with `@var:` syntax for instance variables.
>     - Automatically loads if `amazing_print` is available.
>   - **Pretty Print Support**:
>     - Implements `pretty_print` in `AI::Chat` and `AI::Response` for use with Ruby's `pp`.
>   - **Features**:
>     - Truncates long message content to 80 characters with "...".
>     - Hides sensitive instance variables (`@api_key`, `@client`).
>     - Improves indentation and object notation for debugging.
>   - **Miscellaneous**:
>     - Updates `.gitignore` to exclude temporary demo scripts.
>     - Minor dependency updates in `Gemfile.lock`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=firstdraft%2Fai-chat&utm_source=github&utm_medium=referral)<sup> for 6e63430d663c555bdefe0faeb421b259ddf81d79. You can [customize](https://app.ellipsis.dev/firstdraft/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

raghubetina commented:
Closing to start with just Gemini compatibility instead.

raghubetina committed:
Add custom amazing_print formatting for AI::Chat

Implement to_hash method that provides a clean, readable format when
using amazing_print. Features include:
- Dynamic inclusion of all instance variables
- Security: excludes @api_key and @client
- Truncates long message content to 80 chars
- Skips nil values for cleaner output

raghubetina committed:
Add amazing_print support for AI::Response

Implement to_hash method that dynamically includes all instance
variables for clean formatting. This makes Response objects display
with proper indentation when nested within Chat objects.

raghubetina committed:
Improve amazing_print formatting for AI objects

Replace hash-based formatting with object notation for better clarity.
AI::Chat and AI::Response now display as proper objects with instance
variables using @var: syntax. Features include:

- Object notation (#<AI::Chat ...>) instead of hash format
- Instance variables shown with colons (@messages:, @model:, etc)
- Automatic content truncation for long messages (80 chars)
- Sensitive data filtering (@api_key, @client hidden)
- Proper indentation for nested objects and arrays

This makes the output more intuitive for beginners who expect to see
objects rather than hash representations.

jelaniwoods commented:
This is neat. I didn't realize how much you could do with amazing print. Did you mention previously that you didn't like including the `[...]` in from of Array elements?

raghubetina committed:
Add demo2/ to .gitignore

Exclude temporary demo scripts from version control

raghubetina committed:
Add pretty_print support for Ruby's pp library

Implement pretty_print methods for AI::Chat and AI::Response to provide
clean output with Ruby's built-in pp command. Features:
- Truncates long message content to 80 characters
- Hides sensitive instance variables (@api_key, @client)
- Displays objects with proper notation and indentation
- Works alongside amazing_print for maximum flexibility

raghubetina commented:
> This is neat. I didn't realize how much you could do with amazing print. Did you mention previously that you didn't like including the `[...]` in from of Array elements?

I think I'm okay with it in this context. And, if we start adding an `.amazing_print.yml` to codespaces to eliminate that in general, it might go away here too — not sure.

raghubetina commented:
#10 should make this a bit simpler, if we decide to go that way.

jelaniwoods commented:
@raghubetina aside from addressing merge conflicts and code quality, is this PR feature complete?

jelaniwoods committed:
Merge main

jelaniwoods committed:
Resolve inconsistencies
